### PR TITLE
Morebuttons

### PR DESCRIPTION
**Pull request by martinhoefling**
Adds buttons on hover as alternate way to clipboard copy and detail view.

### CODE REVIEW

**Comment by martinhoefling** [comment]
@pharb, shall I write tests for it or shall I wait to reduce the risk of conflicts?

**Comment by Pharb** [comment]
@martinhoefling Please feel free to push anything you want 😃
I will always try to have everything fully pushed to PRs and if necessary I will keep a "WIP" PR open if I have something in progress.

File: web-extension/search.js [hunk 1]
```diff
@@ -121,26 +127,39 @@ function _onSearchResults(response, isHostQuery) {
...
-        );
+        _onEntryDetails(event);
     } else if (event.shiftKey) {
-        sendNativeAppMessage({ type: 'getLogin', entry: value }).then(
-            _onLoginCredentialsDoCopyClipboard,
-            logAndDisplayError
-        );
+        _onEntryCopy(event);
     } else {
+        const value = element.innerText;
```
Comment (Pharb) [inline web-extension/search.js]: I'm getting an error here when trying to login `search.js, line 140: TypeError: element is undefined`

File: web-extension/styles.css [hunk 2]
```diff
@@ -87,6 +91,45 @@ body {
...
+}
+
+.entry .details {
+    right: 5px;
+    background-image: url('icons/si-glyph-arrow-thick-thin-down.svg');
+}
+
+.entry .copy {
+    right: 35px;
+    background-image: url('icons/si-glyph-document-copy.svg');
```
Comment (Pharb) [inline web-extension/styles.css]: This icon doesn't work for me.

File: web-extension/search.js [hunk 3]
```diff
@@ -121,26 +127,39 @@ function _onSearchResults(response, isHostQuery) {
     searching = false;
 }
 
-function _onResultSelected(event, element) {
-    element = element || event.target;
-    const value = element.innerText;
+function _onEntryAction(event, element) {
+    if (element) {
+        event.target = element;
```
Comment (Pharb) [inline web-extension/search.js]: This won't work. `event.target` is a getter [according to the DOM spec](https://dom.spec.whatwg.org/#dom-event-target). Setting it is silently ignored in Chrome and throws and error in Firefox: `TypeError: setting getter-only property "target"`.

File: web-extension/search.js [hunk 4]
```diff
@@ -121,26 +127,39 @@ function _onSearchResults(response, isHostQuery) {
...
 
+function _onEntryCopy(event) {
+    const value = event.target.innerText;
+    sendNativeAppMessage({ type: 'getLogin', entry: value }).then(
+        _onLoginCredentialsDoCopyClipboard,
+        logAndDisplayError
+    );
+}
+
+function _onEntryDetails(event) {
```
Comment (Pharb) [inline web-extension/search.js]: _onEntryCopy and _onEntryDetails don't need the full event, maybe just pass the "target" element explicitly?

File: web-extension/_locales/en/messages.json [hunk 5]
```diff
@@ -88,6 +88,10 @@
     "message": "To copy the secret to clipboard, hold shift key while clicking on the secret.",
     "description": "Clipboard hint text"
   },
+  "optionsUsageHintsDetailsText": {
+    "message": "To open the metadate for an entry, hold alt key while clicking on the secret.",
```
Comment (Pharb) [inline web-extension/_locales/en/messages.json]: Typo "metadate". Also I don't think "metadata" is a common enough word to use here.
Maybe just "To show details for an entry [...]".

File: web-extension/_locales/de/messages.json [hunk 6]
```diff
@@ -85,9 +85,13 @@
     "description": "Title of UsageHints"
   },
   "optionsUsageHintsClipboardText": {
-    "message": "Um ein Passwort in die Zwischenablage zu kopieren, Großstelltaste halten während man auf den Eintrag klickt..",
+    "message": "Um ein Passwort in die Zwischenablage zu kopieren, Großstelltaste halten während man auf den Eintrag klickt",
     "description": "Clipboard hint text"
   },
+  "optionsUsageHintsDetailsText": {
+    "message": "Um die Metadaten des Eintrages anzuzeigen, Alt Taste halten während man auf den Eintrag klickt",
```
Comment (Pharb) [inline web-extension/_locales/de/messages.json]: Maybe "Um den vollen Eintrag anzuzeigen [...]"

**Comment by martinhoefling** [comment]
I hope this fixes the issues. The button design is very ugly atm but my design skills are very limited.

**Pull request merged**